packages/job-manager/src/worker/worker.ts

```diff
@@ -65,12 +66,24 @@ export class Worker {
                 break;
             }
 
-            if (moment().toDate() > this.restartAfterTime) {
+            if (moment().toDate() >= this.restartAfterTime) {
                 this.logger.logInfo(`Performing scheduled termination after ${this.jobManagerConfig.maxWallClockTimeInHours} hours.`);
+                await this.waitForChildTasks();
                 break;
             }
 
-            await System.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+            await this.system.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+        }
+    }
+
+    private async waitForChildTasks(): Promise<void> {
+        this.logger.logInfo('Waiting for child tasks to complete');
+
+        let poolMetricsInfo: PoolMetricsInfo = await this.batch.getPoolMetricsInfo();
+
+        while (poolMetricsInfo.load.activeTasks !== 0 || poolMetricsInfo.load.runningTasks > 1) {
```

Comment:
Don't you think the termination condition should be 
if (poolMetricsInfo.load.activeTasks === 0 && poolMetricsInfo.load.runningTasks === 1) as per line 42?

Reply:
both are same. (negation of what i have) in this case, i keep waiting if there are any active task or running task